editor/gui/create_dialog.cpp

```diff
@@ -222,7 +226,7 @@ bool CreateDialog::_should_hide_type(const StringName &p_type) const {
 		// Abstract scripts cannot be instantiated.
 		String path = ScriptServer::get_global_class_path(p_type);
 		Ref<Script> scr = ResourceLoader::load(path, "Script");
-		return scr.is_null() || scr->is_abstract();
+		return scr.is_null() || (scr->is_abstract() && !allow_abstract_scripts);
```

Comment:
```suggestion
		return scr.is_null() || (!allow_abstract_scripts && scr->is_abstract());
```
Cheaper checks first.